src/main/java/com/demo/api/service/StaffService.java

```diff
@@ -5,16 +5,79 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import com.demo.api.model.Staff;
+import com.demo.api.entities.StaffEntity;
+import com.demo.api.model.response.StaffResponse;
+import com.demo.api.model.response.StaffsResponse;
 import com.demo.api.repositories.StaffRepository;
 
+/**
+ * Staff Service.
+ * 
+ * @author QuynhNN
+ */
 @Service
 public class StaffService {
 
     @Autowired
     private StaffRepository staffRepository;
 
-    public List<Staff> findAll() {
-        return this.staffRepository.findAll();
+    /**
+     * get all staff list.
+     * 
+     * @return staffs response list.
+     */
+    public StaffsResponse getAllStaff() {
+        List<StaffEntity> staffList = this.staffRepository.getAll();
+        Mapper mapper = new Mapper(staffList);
+        StaffsResponse response = new StaffsResponse();
+        response.setStaffs(mapper.map());
+        return response;
     }
+
+    /**
+     * get staff details by id.
+     * 
+     * @param id the id of the staff.
+     * @return staff's information.
+     */
+    public StaffResponse getStaffDetailsById(int id){
+        StaffEntity staffEntity = this.staffRepository.getStaffDetailsById(id);
+        StaffResponse staffResponse = new StaffResponse();
+        staffResponse.setId(staffEntity.getId());
+        staffResponse.setName(staffEntity.getName());
+        staffResponse.setAddress(staffEntity.getAddress());
+        staffResponse.setPhoneNumber(staffEntity.getPhoneNumber());
+        staffResponse.setDateOfBirth(staffEntity.getDateOfBirth());
+        return staffResponse;
+    }
+
+    /**
+     * Update staff by id.
+     * 
+     * @param staff new information of staff
+     * @param id the staff id
+     */
+    public void updateStaffById(StaffResponse staff, int id) {
+        StaffEntity staffEntity = new StaffEntity();
+        staffEntity.setName(staff.getName());
+        staffEntity.setAddress(staff.getAddress());
+        staffEntity.setPhoneNumber(staff.getPhoneNumber());
+        staffEntity.setDateOfBirth(staff.getDateOfBirth());
+        this.staffRepository.updateStaffById(staffEntity, id);
```

Comment:
trường hợp user có id đang update đã bị xóa khỏi DB thì sẽ như thế nào

Reply:
mình cần phải thông báo cho user biết id không tồn tại

Reply:
em báo id không tồn tại
public void updateStaffById(StaffRegisterRequest staffRegisterRequest){
        if (!staffRepository.findById(staffRegisterRequest.getId())){
            throw new BadRequestException(new ApiError("id_not_found", "ID not found."));
        }

        int id = staffRegisterRequest.getId();
        if (!staffRepository.findById(id)) {
            throw new IdNotFoundException();
        } else {
            StaffEntity staffEntity = new StaffEntity();
            staffEntity.setId(staffRegisterRequest.getId());
            staffEntity.setName(staffRegisterRequest.getName());
            staffEntity.setAddress(staffRegisterRequest.getAddress());
            staffEntity.setPhoneNumber(staffRegisterRequest.getPhoneNumber());
            staffEntity.setDateOfBirth(staffRegisterRequest.getDateOfBirth());
            this.staffRepository.updateStaffById(staffEntity, staffEntity.getId());
        }
    }